sklego/meta/hierarchical_predictor.py

```diff
@@ -220,13 +222,17 @@ def __init__(
         self.n_jobs = n_jobs
         self.check_X = check_X
         self.shrinkage_kwargs = shrinkage_kwargs
+        try:
+            self.estimator_supports_sample_weight = "sample_weight" in inspect.signature(self.estimator.fit).parameters
+        except Exception:
+            self.estimator_supports_sample_weight = False
```

Comment:
For a scikit-learn compatible estimator you cannot do any operation in `__init__` method. You can assign such attribute only during fit, and it must be (semi) private, with a trailing `_`